RubberduckTests/Inspections/ImplicitDefaultMemberAccessInspectionTests.cs

```diff
@@ -0,0 +1,181 @@
+        //There is a specialized inspection for this.
+        public void ProcedureCoercion_NoResult()
+        {
+            var class1Code = @"
+Public Function Foo() As Long
+Attribute Foo.VB_UserMemId = 0
+End Function
+";
+
+            var moduleCode = $@"
+Private Function Foo() As Long 
+    Dim cls As New Class1
+    cls
+End Function
+";
+
+
+            var vbe = MockVbeBuilder.BuildFromModules(
+                ("Class1", class1Code, ComponentType.ClassModule),
+                ("Module1", moduleCode, ComponentType.StandardModule));
+
+            var inspectionResults = InspectionResults(vbe.Object);
+
+            Assert.AreEqual(0, inspectionResults.Count());
+        }
+
+        [Test]
+        [Category("Inspections")]
+        [TestCase("Foo = cls")]
+        [TestCase("cls = bar")]
+        //There is a specialized inspection for this.
+        public void RecursiveImplicitDefaultMemberAccessExpression_NoResult(string statement)
+        {
+            var class1Code = @"
+Public Function Foo() As Class2
+Attribute Foo.VB_UserMemId = 0
+End Function
+";
+
+            var class2Code = @"
+Public Function Baz() As Long
+Attribute Baz.VB_UserMemId = 0
+End Function
+";
+
+            var moduleCode = $@"
+Private Function Foo() As Long 
+    Dim cls As New Class1
+    Dim bar As Long
+    {statement}
+End Function
+";
+
+            var vbe = MockVbeBuilder.BuildFromModules(
+                ("Class1", class1Code, ComponentType.ClassModule),
+                ("Class2", class2Code, ComponentType.ClassModule),
+                ("Module1", moduleCode, ComponentType.StandardModule));
+
+            var inspectionResults = InspectionResults(vbe.Object);
+
+            Assert.AreEqual(0, inspectionResults.Count());
+        }
+
+        [Test]
+        [Category("Inspections")]
+        [TestCase("Foo = cls(0)")]
+        [TestCase("cls(0) = bar")]
+        public void OrdinaryImplicitDefaultMemberAccessOnDefaultMemberArrayAccess_TwoResults(string statement)
+        {
+            var class1Code = @"
+Public Function Foo() As Class2()
+Attribute Foo.VB_UserMemId = 0
+End Function
+";
+
+            var class2Code = @"
+Public Function Baz() As Long
+Attribute Baz.VB_UserMemId = 0
+End Function
+";
+
+            var moduleCode = $@"
+Private Function Foo() As Long 
+    Dim cls As New Class1
+    Dim bar As Long
+    {statement}
+End Function
+";
+
+            var vbe = MockVbeBuilder.BuildFromModules(
+                ("Class1", class1Code, ComponentType.ClassModule),
+                ("Class2", class2Code, ComponentType.ClassModule),
+                ("Module1", moduleCode, ComponentType.StandardModule));
+
+            var inspectionResults = InspectionResults(vbe.Object);
+
+            Assert.AreEqual(2, inspectionResults.Count());
+        }
+
+        protected override IInspection InspectionUnderTest(RubberduckParserState state)
+        {
+            return new ImplicitDefaultMemberAccessInspection(state);
```

Comment:
Could be expression-bodied (nit)

Reply:
TBH, I prefer to use expression bodied members only for properties. If you want me to turn this into an expression bodied meber, just poke me again.